lib/primitives/elementary/point.rb

```diff
@@ -0,0 +1,53 @@
+module Primitives
+  module Elementary
+    class Point
+
+      def check_argument(arg)
+        unless arg.is_a?(Integer)
+          raise TypeError.new "Invalid type of argument (Must be Int)"
+        end
+      end
+
+      def initialize(x, y)
+        check_argument(x)
+        check_argument(y)
+        @x = x
+        @y = y
+      end
+
+      # Changes point's position
+      def move!(offset_x, offset_y)
+        self = self.move(offset_x, offset_y)
+      end
+
+      # Changes point's position
+      def move(offset_x, offset_y)
+        check_argument(offset_x)
+        check_argument(offset_y)
+        return Point.new(@x + offset_x, @y + offset_y)

```

Comment:
There is no need to call `return` explicitly because Ruby able to return the value of the last executed instruction in the method. It`s not a problem but if a statement may be ignored it means that the explicitly statement specification is redundant.